add initial contributing guidelines (#338)
neontribe/www: CONTRIBUTING.md

# Contribution guidelines

Ongoing development on this project is managed by this project board: https://github.com/neontribe/www/projects/6.

## Getting started on development

> Any issues in the `To do` column are assumed to be important enough and have been prioritised.

Reviewer: Is going to need changing if today's meeting approves the removal of the "to do" column

When contributing to this project please do the following:

> 1. Pick something near the top of the to do column, which you feel you have the time and confidence to tackle.

Reviewer 2: ```suggestion
1. Pick something near the top of the `To do` column, which you feel you have the time and confidence to tackle.
```

Consistency :-)

   - We expect you to learn while tackling issues, please don't be put off by a tasks' seeming complexity.

>    - If you need more information please follow up with the person who spawned the task.

Reviewer: ```suggestion
   - If you need more information please follow up with the person who spawned/created the task.
```
I almost went for a change to plain English and got rid of spawned - but then thought actually, no its good for people to learn coders language as they start working on this - what do people feel about using both the coders slang term and the plain english term for things like this?

   - Check the [Definition of done](#definition-of-done) for the ticket

> 1. Assign yourself and move the ticket to the In progress column.

Reviewer 2: ```suggestion
1. Assign yourself and move the ticket to the `In progress` column.
```

1. Create a branch with a name to help match it to the ticket, e.g.: `307-contrib-docs`. _n.b._ [avoid slashes in branch names](https://stackoverflow.com/questions/2527355/using-the-slash-character-in-git-branch-name/2527452#2527452)

> 1. Work on the feature branch, make small commits which include the _reason_ for each commit's change.

Reviewer 3: Maybe a tip - don't forget to include tests if appropriate

> 1. Work on the feature branch, make small commits which include the _reason_ for each commit's change.

Reviewer 3: Additionally, I like these: https://chris.beams.io/posts/git-commit/

Probably not necessary to include that specific link but the most important one for me is:
https://chris.beams.io/posts/git-commit/#imperative

1. Create a pull request when you're happy with the changeset, and happy that the ticket passes the ticket's [Definition of done](#definition-of-done).

>    _n.b._ Pull requests are the pull requesters responsibility, when it recieves the request number

Reviewer 4: ```suggestion
   _n.b._ Pull requests are the pull requester's responsibility, when it receives the request number
```

1. Await code review, see [Review proces](#review-process)

> Please make sure that you keep you activity up-to-date:

Reviewer 2: ```suggestion
Please make sure that you keep your activity up-to-date:
```

Or maybe that's not needed at all? As in, you've added the word 'you' twice?

> - only assign one ticket at a time to yourself

Reviewer 3: Maybe we can incorporate these 3 into the when contributing list? e.g. the branch name is already there.

- name a branch based upon the ticket

> - if the ticket needs to be put on hold

Reviewer 2: seems like an unfinished point - I see an `if` I expect a then or an else :smiley:

Reviewer: Is the second half an instruction for how to label if it has been put on hold?

---

> WIP:

Reviewer 3: Could tidy the WIP section grammatically with uppercase letters and lists even where only one item - for better readability/ consistency

## Getting tickets ready to be worked on

- weekly meeting to discuss open tickets
- tickets should be left in the backlog or closed if vetoed
- anyone can create issues

> - creating an issue involves writing up a compelling argument for a feature, perhaps pictures. Adding a definition of done is important, doesn't have to be acceptance criteria

Reviewer: ```suggestion
- creating an issue involves writing up a compelling argument for an improvement, perhaps including pictures if it is a change to the existing site. Anything new should be supported by a user story that will help to inform a definition of done. This definition of done is important, but does not have to be full acceptance criteria at this stage.
```
It had the word feature in. A little part of me cried. Have I taken it too far the other way?

## Review process

> 2 or more peers need to approve the commit, when reviewing check the code for errors and check that it achieves the definition of done for the ticket

Reviewer 4: ```suggestion
Two or more peers need to approve the commit, when reviewing check the code for errors and check that it achieves the definition of done for the ticket
```

Reviewer 2: To me, these are in fact 3 separate points that can be split over three lines (would improve readability, too):
```
- Two or more peers need to approve the commit 
- When reviewing, check the code for errors 
- Check that it achieves the definition of done for the ticket
```

## Testing

is done using feature branch deploys on pull request

## Definition of done

> Acceptance critera etc.

Reviewer 4: ```suggestion
Acceptance criteria etc.
```
